[Ready for Review] Update BinderHub workshop (#1552)
alan-turing-institute/the-turing-way: workshops/build-a-binderhub/workshop-presentations/zero-to-binderhub.md

> Make a folder to store this files and change into it.

Reviewer: typo: 'this' should be 'the' or 'these'.

> Make a folder to store this files and change into it.

Author: ```suggestion
Make a folder to store these files and change into it.
```

> Make the shell script executable will the following command.

Reviewer: typo: 'will' should be 'with'

> Make the shell script executable will the following command.

Author: ```suggestion
Make the shell script executable with the following command.
```

Adapted from [Step Zero: Kubernetes on Microsoft Azure Container Service (AKS)](https://zero-to-jupyterhub.readthedocs.io/en/latest/microsoft/step-zero-azure.html).

A short (but by no means exhaustive) [glossary](#glossary-of-kubernetes-terms) of Kubernetes terms is given at the end of this workshop, should you require further explanation.

> #### 1. Login to your account

Author: This is wrong ! Please change!

Run the following command:

```bash

> az login

Author: ```suggestion
az log-in
```

To see a list of Azure subscriptions you have available to you, you can run the following command.

```bash
az account list --refresh --output table
```

This prints your subscriptions to the terminal in a human-readable format.
Now let's set our working subscription.

**If you are using a Free Trial suscription:**

```bash
az account set --subscription "Free Trial"
```

**If you are using an Azure Pass:**

```bash
az account set --subscription "Azure Pass - Sponsorship"
```

:vertical_traffic_light: :vertical_traffic_light: :vertical_traffic_light: :vertical_traffic_light: :vertical_traffic_light:

**NOTE:** If your subscription name has no whitespace, the quotation marks are not required.

#### 3. Create a Resource Group

Resource Groups are how the Azure environment labels services that are related to each other (further details in [this blog post](http://www.onlinetech.com/resources/references/how-to-use-azure-resource-groups-a-simple-explanation)).
We will create a resource group in a specific region and create computational resources _within_ this group.

```bash
az group create --name testhub \
    --location westeurope \
    --output table
```

> * `--name` specifies the name of your resource group and should be something that uniquely identifies this hub.

Reviewer: Potential source of confusion? "uniquely identifies this hub".  Multiple people in a workshop will be using the same name `testhub` which suggests you do not need to have a unique name. I do not know anything about Azure so I am not sure. A possible re-wording:
"...name of your resource group. In practice this should be something that uniquely identifies this hub, but for this workshop, we will all use the same name `testhub`"

> * `--name` specifies the name of your resource group and should be something that uniquely identifies this hub.

Author: ```suggestion
* `--name` specifies the name of your resource group and should be something that uniquely identifies this hub from other hubs you may deploy.
```

:vertical_traffic_light: :vertical_traffic_light: :vertical_traffic_light: :vertical_traffic_light: :vertical_traffic_light:

#### 4. Create an Azure Kubernetes Service (AKS) Cluster

This command will request a Kubernetes cluster within the resource group we created.

> It will request one `Standard_D2s_v3` virtual machine which a Kubernetes cluster installed.

Reviewer: typo, I think: 'which' should be 'with'

> It will request one `Standard_D2s_v3` virtual machine which a Kubernetes cluster installed.

Author: ```suggestion
It will request a Kubernetes cluster with one `Standard_D2s_v3` virtual machine as a 'node'.
```

For the sake of this workshop, you should have created your account with a different email address.
Otherwise, you could ask your IT Services to provide you with a Service Principal.

If your Cloud Shell has timed out while the cluster was creating, you can do the following steps to retrieve the logs and see if the deployment was successful.

#### A. Locate your Resource Groups

> In Azure Portal, select "Resource Groups from the left-most menu panel.

Reviewer: typo: missing quotation mark

> In Azure Portal, select "Resource Groups from the left-most menu panel.

Author: ```suggestion
In Azure Portal, select "Resource Groups" from the left-most menu panel.
```

![Find Resource Groups](../binderhub_resources/resource_group1.png)

#### B. Select `testhub`

From the list of resource groups, select `testhub`.

![Select testhub Resource Group](../binderhub_resources/resource_group2.png)

#### C. Select the Activity Log pane

From the left-side menu, select "Activity log".

![Select Activity Log](../binderhub_resources/activity_log1.png)

#### D. View the logs for Managed Cluster

> The logs for deploying the cluster will be under "Managed Cluster".

Reviewer: Suggestion: say something about what they are looking for in the logs. E.g. that Status says "Suceeded".  If you do decide to make this change, then maybe change the accompanying image to have box around the whole "Managed Cluster"'s row, or have arrows pointing to "Managed Cluster" and to "Suceeded"

### Setting up Helm

Adapted from [Zero-to-JupyterHub: Setting up Helm](https://zero-to-jupyterhub.readthedocs.io/en/latest/kubernetes/setup-helm.html).

Helm is the package manager for Kubernetes and is used for installing, upgrading and managing applications on a Kubernetes cluster.
Helm packages are called _charts_.
Helm manages _releases_ (installations) and _revisions_ (versions) of charts deployed on the cluster.

> **Did you know?:** Kubernetes is Greek for "captain" or "helmsman".
> In case you haven't noticed the nautical theme!

> #### 1. Verify Helm

Reviewer: In jupyterhub resource, they have installation step before verification.

Author: Yes, they do. I normally run this workshop using the Azure Cloud Shell however, which has a lot of tools preinstalled so I don't have to worry about getting people setup before the day.

Next, install the required Helm chart using the config files we created in [Step 2: Run `setup.sh`](#2-run-setupsh).

```bash
helm install binderhub jupyterhub/binderhub \
    --version=0.2.0-f565958 \
    --namespace=binderhub \
    -f secrets/secret.yaml \
    -f secrets/config.yaml \
    --create-namespace
```

* `--version` refers to the version of the BinderHub Helm Chart.
  Available versions can be found [here](https://jupyterhub.github.io/helm-chart/#development-releases-binderhub).
  We have used the version released on 11 August 2019.

> * It is recommended that `--namespace` be the same as the provided `NAME` in order to avoid confusion.

Reviewer: Where is `NAME` provided? Possible I just missed it.

Author: `NAME` is `binderhub`. The Helm CLI has updated so NAME is now a positional argument instead of an optional arg with a flag.

> * It is recommended that `--namespace` be the same as the provided `NAME` in order to avoid confusion.

Author: ```suggestion
* It is recommended that `--namespace` be the same as the provided `NAME` in order to avoid confusion.
  In this case, our `NAME` is `binderhub`.
```
